iron-overlay-manager.html

```diff
@@ -12,60 +12,74 @@
 
 <script>
 
-  Polymer.IronOverlayManager = (function() {
+  Polymer.IronOverlayManager = {
 
-    var overlays = [];
-    var DEFAULT_Z = 10;
-    var backdrops = [];
+    _overlays: [],
+
+    // iframes have a default z-index of 100, so this default should be at least
+    // that.
+    _minimumZ: 101,
+
+    _backdrops: [],
 
     // track overlays for z-index and focus managemant
-    function addOverlay(overlay) {
-      var z0 = currentOverlayZ();
-      overlays.push(overlay);
-      var z1 = currentOverlayZ();
-      if (z1 <= z0) {
-        applyOverlayZ(overlay, z0);
+    addOverlay: function(overlay) {
+      var minimumZ = Math.max(this.currentOverlayZ(), this._minimumZ);
```

Comment:
`currentOverlayZ` defaults to `_minimumZ` of there are no overlays, but if there are overlays it returns the z-index of the last opened overlay (I actually wonder about that behavior, but it's what was there). In cases where `ensureMinimumZ` is called after an overlay has been added, with a value greater than the current Z, then `currentOverlayZ` will return a value less than `_minimumZ`.